src/general/StringUtils.cs

```diff
@@ -41,4 +44,116 @@ public static string FormatNumber(this double number, bool withSuffix = true)
+
+            if (validQuotes && !insideQuote && ignoreWithinQuotes &&
+                input.IndexOf("\"", i + 1, StringComparison.InvariantCulture) != -1)
+            {
+                insideQuote = true;
+            }
+            else if (validQuotes && insideQuote)
+            {
+                insideQuote = false;
+            }
+
+            if (character == ' ' && !insideQuote)
+            {
+                if ((i == 0) || (i + 1 < input.Length && input[i + 1] == ' '))
+                {
+                    cutPosition++;
+                    continue;
+                }
+
+                result.Add(input.Substring(cutPosition, i - cutPosition).Unescape());
+                cutPosition = i + 1;
+            }
+
+            // Reached end of string, add the rest of it from last cut point (whitespace)
+            if (i == input.Length - 1)
+                result.Add(input.Substring(cutPosition, i - cutPosition + 1).Unescape());
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    ///   Parses a list of "key=value" pairs into a dictionary. Duplicate keys will be skipped.
+    /// </summary>
+    /// <returns>
+    ///   A dictionary of key and value string pairs collected from input. If input string list is null,
+    ///   the return value is an empty dictionary.
+    /// </returns>
+    public static Dictionary<string, string> ParseKeyValuePairs(List<string> input)
+    {
+        var result = new Dictionary<string, string>();
+
+        if (input == null)
+            return result;
+
+        foreach (var entry in input)
+        {
+            if (entry.IndexOf("=", StringComparison.InvariantCulture) == -1)
+                continue;
+
+            var split = entry.Split("=");
+
+            if (split.Length != 2)
+                continue;
+
+            if (!result.ContainsKey(split[0]))
+                result.Add(split[0], split[1]);
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    ///   Checks and returns true if the input string starts and ends with the given string.
+    /// </summary>
+    public static bool IsEnclosedIn(this string input, string what)
+    {
+        return input.StartsWith(what, StringComparison.InvariantCulture) &&
+            input.EndsWith(what, StringComparison.InvariantCulture);
+    }
+
+    /// <summary>
+    ///   Converts any escaped characters in the input string.
+    /// </summary>
+    /// <returns>A string with any escaped characters replaced by their unescaped form.</returns>
+    public static string Unescape(this string input)
+    {
+        var result = new StringBuilder(input);
+
+        result.Replace("\\a", "\a")
+            .Replace("\\b", "\b")
+            .Replace("\\f", "\f")
+            .Replace("\\n", "\n")
+            .Replace("\\r", "\r")
+            .Replace("\\t", "\t")
+            .Replace("\\v", "\v")
```

Comment:
Well I assume the other sequences would do what they represent since newline, double quotes and backslash works correctly. I guess another static method could be added just for unescaping the whitespace controls?

Reply:
Because I think we might use rich text for stuff like user input (in game chat for multiplayer if we add that), it's absolutely necessary that by *default* these bell and backspace escape codes are not allowed.

Reply:
Removed some of the sequences.